### PR TITLE
Feat/2782 project crud

### PR DESCRIPTION
**Pull request by abrl91**
Issue Number: #2782 

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

### CODE REVIEW

**Comment by abrl91** [comment]
@arielweinberger I am tagging you here because I used a different naming convention for the DTOs: 
https://docs.nestjs.com/graphql/resolvers#dedicated-arguments-class

I saw your issue: https://github.com/amplication/amplication/issues/2968
but it seems different than what Nest have on their documentation (regarding args and inputs type)

**Comment by arielweinberger** [comment]
@abrl91 that's fine for Args and InputTypes. My convention issue was about REST DTOs.

**Comment by tupe12334** [comment]
@abrl91 why did you put the project dto in the models folder as the past convention and not in the project->dto folder as the new convention

**Comment by abrl91** [comment]
> @abrl91 why did you put the project dto in the models folder as the past convention and not in the project->dto folder as the new convention

because of all (or most) of our models there. When we will do refactoring, we will change. For now, if someone searches for a project model he/she will find it where all the models

File: packages/amplication-server/src/models/Project.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  })
+  name!: string;
+
+  workspace?: Workspace;
+
+  workspaceId?: string;
+
+  @Field(() => [Resource], {
+    nullable: true,
+    description: undefined
```
Comment (tupe12334) [inline packages/amplication-server/src/models/Project.ts:28]:
```suggestion
```

File: packages/amplication-server/src/models/Project.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export class Project {
+  @Field(() => String, {
+    nullable: false,
+    description: undefined
+  })
+  id!: string;
+
+  @Field(() => String, {
+    nullable: false,
+    description: undefined
```
Comment (tupe12334) [inline packages/amplication-server/src/models/Project.ts:18]:
```suggestion
```

File: packages/amplication-server/src/models/Project.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { Workspace } from './Workspace';
+import { Resource } from './Resource';
+import { Field, ObjectType } from '@nestjs/graphql';
+
+@ObjectType({
+  isAbstract: true,
+  description: undefined
```
Comment (tupe12334) [inline packages/amplication-server/src/models/Project.ts:7]:
```suggestion
```

File: packages/amplication-server/src/models/Project.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+})
+export class Project {
+  @Field(() => String, {
+    nullable: false,
+    description: undefined
```
Comment (tupe12334) [inline packages/amplication-server/src/models/Project.ts:12]:
```suggestion
```

File: packages/amplication-server/src/core/project/dto/project-find-many.args.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  where?: ProjectWhereInput;
+
+  @Field(() => ProjectOrderByInput, { nullable: true })
+  orderBy?: ProjectOrderByInput;
+
+  @Field(() => Int, { nullable: true })
+  skip?: number | null;
+
+  @Field(() => Int, { nullable: true })
+  take?: number | null;
```
Comment (tupe12334) [inline packages/amplication-server/src/core/project/dto/project-find-many.args.ts:17]: see above.

File: packages/amplication-server/src/models/Project.ts (hunk 1 above)
Comment (abrl91) [reply packages/amplication-server/src/models/Project.ts:28]: we talked about it. All the other dto are like that. We will create an issue about it and fix it for all the entities.

File: packages/amplication-server/src/core/project/dto/project-find-many.args.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+@ArgsType()
+export class ProjectFindManyArgs {
+  @Field(() => ProjectWhereInput, { nullable: true })
+  where?: ProjectWhereInput;
+
+  @Field(() => ProjectOrderByInput, { nullable: true })
+  orderBy?: ProjectOrderByInput;
+
+  @Field(() => Int, { nullable: true })
+  skip?: number | null;
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-server/src/core/project/dto/project-find-many.args.ts:14]: we are doing it like that in a lot of places. For example:
image

File: packages/amplication-server/src/models/Project.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+  @Field(() => String, {
+    nullable: false,
+    description: undefined
+  })
+  name!: string;
+
+  workspace?: Workspace;
+
+  workspaceId?: string;
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-server/src/models/Project.ts:24]: Have a look at Resource.ts 
Project now is like Resource, why is it like that in Resource?

File: packages/amplication-server/src/core/project/dto/project-find-many.args.ts (hunk 5 above)
Comment (abrl91) [reply packages/amplication-server/src/core/project/dto/project-find-many.args.ts:17]: same

**Pull request merged**